src/builtins/network.rs

```diff
@@ -201,8 +207,7 @@ impl Networking {
         self.custom.entry(Sysno::socket)
             .or_insert_with(Vec::new)
             .push(rule);
-
```

Comment:
I don't think it makes sense to remove the call here since I can't imagine a case where you'd want to use this method without the connect call.

Reply:
True, however this makes the code look inconsistent from when you want to only allow UDP or Unix socket access. 
I think one thing we could do is perhaps allow `connect()` in all three socket types? Then we could likely get rid of `allow_connect()`.

Reply:
I probably need to refresh myself on sockets but I don't think the APIs themselves are consistent. For TCP only the client needs to connect, but for UDP and Unix sockets connect is entirely optional. In rust it will depend on what network library you're using (and possibly even your libc implementation?). So it might be fine to just add `allow_connect` and then add a note to the UDP and Unix methods that depending on your library you may need to enable connect as well.

I'm also not 100% sure on the security implications of allowing connect always. I think it's *probably* fine but I would like to maybe think about it a bit longer.

It might be good to add an example or test for this as well. E.g. "here's an example of using the raw socket/bind/listen/accept" and "here's an example of using tokio" although the existing ipc tutorials already kind of do that, though maybe not for unix sockets.